pandas/tests/io/test_excel.py

```diff
@@ -2490,3 +2499,82 @@ def custom_converter(css):
             n_cells += 1
 
     assert n_cells == (10 + 1) * (3 + 1)
+
+
+@pytest.mark.parametrize('engine', [
+    pytest.mark.xfail('xlwt', reason='xlwt does not support '
+                                     'openpyxl-compatible style dicts'),
+    'xlsxwriter',
+    'openpyxl',
+])
+def test_styler_to_excel_no_cssdecl(engine, styled_dataframe):
+    def custom_converter(css):
+        # use bold iff there is custom style attached to the cell
+        if css.strip(' \n;'):
+            return {'font': {'bold': True}}
+        return {}
+
+    try:
+        import cssdecl  # noqa
+    except ImportError:
+        pass
+    else:
+        pytest.skip('Test only run if cssdecl not installed')
+
+    pytest.importorskip(engine)
+
+    styled = styled_dataframe
+    df = styled.data
+
+    # Prepare spreadsheets
+
+    with ensure_clean('.xlsx' if engine != 'xlwt' else '.xls') as path:
+        writer = ExcelWriter(path, engine=engine)
+        df.to_excel(writer, sheet_name='frame')
+        with pytest.raises(ImportError):
+            # default style_converter requires cssdecl
+            df.style.to_excel(writer, sheet_name='unstyled')
+            styled.to_excel(writer, sheet_name='styled')
+
+        ExcelFormatter(styled, style_converter=custom_converter).write(
+            writer, sheet_name='custom')
+
+        writer.save()
+
+        openpyxl = pytest.importorskip('openpyxl')
+        if openpyxl_compat.is_compat(major_ver=1):
+            # smoke test only
+            return
+
+        wb = openpyxl.load_workbook(path)
+
+        # check styling with custom converter
+        n_cells = 0
+        for col1, col2 in zip(wb['frame'].columns,
+                              wb['custom'].columns):
+            assert len(col1) == len(col2)
+            for cell1, cell2 in zip(col1, col2):
+                ref = '%s%d' % (cell2.column, cell2.row)
+                if ref in ('B2', 'C3', 'D4', 'B5', 'C6', 'D7', 'B8'):
+                    assert not cell1.font.bold
+                    assert cell2.font.bold
+                else:
+                    assert_equal_style(cell1, cell2)
+
+                assert cell1.value == cell2.value
+                n_cells += 1
+
+        assert n_cells == (10 + 1) * (3 + 1)
+
+
+def test_no_cssdecl_raises():
+    pytest.importorskip("openpyxl")
+    try:
+        import cssdecl  # noqa
+    except ImportError:
+        with pytest.raises(ImportError) as rec:
+            pd.DataFrame({"A": [1, 2]}).style.to_excel("openpyxl")
+
+        assert 'not installed' in rec.value.msg
```

Comment:
Ahh. That's how to do it with pytest. Better here than above (https://github.com/pandas-dev/pandas/pull/16170/files#diff-cd0301639e13a1419d0c4522f08f2be8R2520)?